### PR TITLE
Implement Prim.blobCompare

### LINKED ISSUE #3826: FR: optimize Blob.compare
```
  if (blob1 < blob2) {
      #less
    } else if (blob1 == blob2) {
      #equal
    } else {
      #greater
    }
```
is silly. Ditto Text.compare if not optimized already.

### PR DESCRIPTION
**Pull request by nomeata**
fixes #3826.

### CODE REVIEW

File: src/codegen/compile.ml [hunk 1]
```diff
@@ -3675,29 +3675,40 @@ module Blob = struct
...
+     or implements the generic comparison, returning -1, 0 or 1 as Int8.
+  *)
   let rec compare env op =
+    let word32_cmp get_a get_b =
+        (* Wasm has no compare instruction for Word32 that returns -1, 0, 1. Here is one *)
+        get_a ^^ get_b ^^ G.i (Compare (Wasm.Values.I32 I32Op.LtU)) ^^
+        G.if0 (compile_unboxed_const (TaggedSmallWord.vanilla_lit Type.Int8 (-1)) ^^ G.i Return) G.nop ^^
+        get_a ^^ get_b ^^ G.i (Compare (Wasm.Values.I32 I32Op.GtU)) ^^
+        G.if0 (compile_unboxed_const (TaggedSmallWord.vanilla_lit Type.Int8 1) ^^ G.i Return) G.nop ^^
+        compile_unboxed_const (TaggedSmallWord.vanilla_lit Type.Int8 0) in
```
Comment (ggreif) [inline src/codegen/compile.ml:3689]: Is it really necessary to leave a 0 on the stack? The other two cases just exit the function, staying in it means the comparison gave an equality. (I am assuming this is being in a loop, so it seems useless stack traffic.)

**Review by ggreif — APPROVED** [review]
Just a nit and a question. LGTM.

File: src/codegen/compile.ml (hunk 1 above)
Comment (nomeata) [reply src/codegen/compile.ml]: This should be more straightforward

**Comment by nomeata** [comment]
Odd, mergify doesn't pick up the approving review it seems.

**Comment by nomeata** [comment]
@Mergifyio refresh

**Comment by nomeata** [comment]
Any idea why the merging doesn't work?

**Comment by ggreif** [comment]
> Any idea why the merging doesn't work?

Screenshot 2023-06-04 at 11 27 07

My approval doesn't seem to count. :-(

**Review by ggreif — APPROVED** [review]

**Comment by nomeata** [comment]
I’ve pinged the mergify support chat for help.

**Comment by nomeata** [comment]
Mergify wrote:

> Hi, we are aware of the issue. GitHub has introduced an API breakage in review API this weekend. An enum has started returning unexpected values. The approval was ignored on purpose as the new permission value was unknown. Now we have a workaround in place, I forced the reevaluation of the pull request to unblock it (you can post a comment with `@mergifyio refresh` if another pull request is stuck on the approval).And we opened a GitHub support ticket to understand the API change.

**Pull request merged**